core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -610,14 +611,32 @@ private void registerBuiltinAggregations() {
                 DerivativePipelineAggregationBuilder.AGGREGATION_NAME_FIELD)
                     .addResultReader(DerivativePipelineAggregator::new)
                     .addBucketReader(InternalDerivative::new));
-        registerPipelineAggregation(MaxBucketPipelineAggregationBuilder::new, MaxBucketPipelineAggregationBuilder.PARSER,
-                MaxBucketPipelineAggregationBuilder.AGGREGATION_NAME_FIELD);
-        registerPipelineAggregation(MinBucketPipelineAggregationBuilder::new, MinBucketPipelineAggregationBuilder.PARSER,
-                MinBucketPipelineAggregationBuilder.AGGREGATION_FIELD_NAME);
-        registerPipelineAggregation(AvgBucketPipelineAggregationBuilder::new, AvgBucketPipelineAggregationBuilder.PARSER,
-                AvgBucketPipelineAggregationBuilder.AGGREGATION_NAME_FIELD);
-        registerPipelineAggregation(SumBucketPipelineAggregationBuilder::new, SumBucketPipelineAggregationBuilder.PARSER,
-                SumBucketPipelineAggregationBuilder.AGGREGATION_NAME_FIELD);
+        registerPipelineAggregation(new PipelineAggregationSpec(
+                MaxBucketPipelineAggregationBuilder::new,
+                MaxBucketPipelineAggregationBuilder.PARSER,
+                MaxBucketPipelineAggregationBuilder.AGGREGATION_NAME_FIELD)
+                    .addResultReader(MaxBucketPipelineAggregator::new)
+                    // This bucket is used by many pipeline aggreations.
+                    .addBucketReader(InternalBucketMetricValue.NAME, InternalBucketMetricValue::new));
+        registerPipelineAggregation(new PipelineAggregationSpec(
+                MinBucketPipelineAggregationBuilder::new,
+                MinBucketPipelineAggregationBuilder.PARSER,
+                MinBucketPipelineAggregationBuilder.AGGREGATION_FIELD_NAME)
+                    .addResultReader(MinBucketPipelineAggregator::new)
```

Comment:
Is this right? surely the resultReader would be InternalMetricValue here (although this is a registered elsewhere so wouldn't need to be explicitly registered here)?